src/main/java/io/cdap/plugin/gcp/dataplex/common/config/DataplexBaseConfig.java

```diff
@@ -0,0 +1,242 @@
+   *
+   * @return true if the service account is set to auto-detect but it can't be fetched from the environment.
+   */
+  public boolean autoServiceAccountUnavailable() {
+    if (connection == null || connection.getServiceAccountFilePath() == null &&
+      connection.isServiceAccountFilePath()) {
+      try {
+        ServiceAccountCredentials.getApplicationDefault();
+      } catch (IOException e) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * @return GoogleCredentials based on service account
+   */
+  public GoogleCredentials getCredentials() {
+    GoogleCredentials credentials = null;
+    try {
+      credentials = getCredentialsFromServiceAccount();
+    } catch (IOException e) {
+      LOG.debug("Unable to load service account credentials due to error: {}", e.getMessage());
+    }
+    return credentials;
+  }
+
+  /**
+   * @return Project ID
+   */
+  public String getProject() {
+    if (connection == null) {
+      throw new IllegalArgumentException(
+        "Could not get project information, connection should not be null!");
+    }
+    return connection.getProject();
+  }
+
+  public GCPConnectorConfig getConnection() {
+    return connection;
+  }
+
+  @Nullable
+  public String tryGetProject() {
+    return connection == null ? null : connection.tryGetProject();
+  }
+
+  @Nullable
+  public String getServiceAccount() {
+    return connection == null ? null : connection.getServiceAccount();
+  }
+
+  @Nullable
+  public String getServiceAccountType() {
+    return connection == null ? null : connection.getServiceAccountType();
+  }
+
+  @Nullable
+  public Boolean isServiceAccountFilePath() {
+    return connection == null ? null : connection.isServiceAccountFilePath();
+  }
+
+  /**
+   * Check if service account is a valid one.
+   *
+   * @param failureCollector
+   */
+  public void validateServiceAccount(FailureCollector failureCollector) {
+    if (connection.isServiceAccountJson() || connection.getServiceAccountFilePath() != null) {
+      GoogleCredentials credentials = getCredentials();
+      if (credentials == null) {
+        failureCollector.addFailure(String.format("Unable to load credentials from %s.",
+              connection.isServiceAccountFilePath() ? connection.getServiceAccountFilePath() : "provided JSON key"),
+            "Ensure the service account file is available on the local filesystem.")
+          .withConfigProperty("serviceFilePath")
+          .withConfigProperty("serviceAccountJSON");
+        throw failureCollector.getOrThrowException();
+      }
+    }
+  }
+
+  private GoogleCredentials getCredentialsFromServiceAccount() throws IOException {
+    GoogleCredentials credentials = null;
+    // validate service account
+    if (connection.isServiceAccountJson() || connection.getServiceAccountFilePath() != null) {
+      credentials =
+        GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath());
```

Comment:
This method expects non-null input. `getServiceAccount()` can still be null since `getServiceAccount()` can return null

Reply:
getServiceAccount will never be null as we have already checked that in if condition

Reply:
`getServiceAccount()` internally calls `connection.getServiceAccount()` which returns null if  `getServiceAccountJson()` is null or empty.

Reply:
handled that